### PR TITLE
Add background elevation semantic tokens

### PR DESCRIPTION
**Pull request by janogarcia**
@gsouquet There are a lot of small commits without really helpful commit messages, as I had to quickly edit things on GitHub.

This PR adds the necessary architecture for supporting semantics tokens per theme. This is required for scenarios where we can't rely on 1:1/symmetrical mappings of tokens across themes, as briefly [explained in PR28](https://github.com/vector-im/compound-design-tokens/pull/28#issuecomment-1485306325).

It also refactors the neutral background styles to add support for elevation. 

🔴 Potential breaking change for consumers: We had to rename the neutral background styles (e.g., `color.bg.canvas` to `color.bg.canvas.elevation-default`) as Tokens Studio won't allow you to have both `color.bg.canvas` and `color.bg.canvas.elevation-1`, where `canvas` is simultaneously the name for a token and the name for a group of tokens.

### CODE REVIEW

File: tokens/$themes.json
Comment (gsouquet) [inline tokens/$themes.json:1]: I'm slightly worried about the huge changeset here.
Do you know why there are so many updates?

**Review by gsouquet — CHANGES_REQUESTED** [review]
Otherwise looks good to me 👍

**Comment by janogarcia** [comment]
@gsouquet On the "Potential breaking change for consumers", we're going to hit that limitation many times when extending and iterating on tokens. For example, let's say we have initially defined a `color.bg.action` and have started using it in Figma and consuming it in the codebases. Later on, we formalize more semantic states for that token (hover, pressed, disabled...), but those now will introduce a naming clash:

- `color.bg.action` — We can't have this token name anymore,  as `.action` is now the name of a token group. It should be renamed to `color.bg.action.default`, which introduces a breaking change for consumers in the codebases.
- `color.bg.action.hover`
- `color.bg.action.pressed`
- `color.bg.action.disabled`

Alternatively, we could do something like this to prevent those clashes between existing token names and new groups that need to be created:

- `color.bg.action`
- `color.bg.action-hover`
- `color.bg.action-pressed`
- `color.bg.action-disabled`

But it's impossible to predict how we're going to need to extend tokens in the future, so we may end up with tokens like `color.bg.action-hover-on-dark`, chaining more modifiers because we don't want to break the existing names.

This is a limitation of both Tokens Studio and the [W3C Design Tokens Format Module](https://design-tokens.github.io/community-group/format/#groups), where you can't have a token and a token group using the same name.

**Comment by gsouquet** [comment]
So in the above, what I would be interested to understand is the transition path we can provide when we face this kind of style reference change. Because you're right, it will ultimately happen and we need to have an answer for that.

But in this scenarion the changeset in `$themes.json` that hosts the figma references seem much larger than what has actually been changed in the tokens source files. Which is surprising. I would expect to retain the style reference if there are no changes to the token.
That bein said, I probably missed one change that yielded this result

File: tokens/$themes.json
Comment (janogarcia) [reply tokens/$themes.json:1]: The Figma style IDs are actually preserved, the reason for shuffling the contents in that files might be that we had to enable a new token set per theme.

So, for example, for the `Light` theme in Tokens Studio, we had to create a new `theme-light-semantics` token set and enable it along with the `theme-light` and `theme-semantics` token sets for the `Light` theme. The style IDs don't change, but the JSON structure for the theme gets updated and maybe even moved to a different location in the file.

**Comment by janogarcia** [comment]
Closing in favor of #33

**Pull request closed without merging**